src/D2L.Bmx/Okta/OktaApi.cs

```diff
@@ -92,44 +101,55 @@ internal class OktaApi : IOktaApi {
-
-		var authnResponse = await JsonSerializer.DeserializeAsync(
-			await resp.Content.ReadAsStreamAsync(),
-			SourceGenerationContext.Default.AuthenticateResponseRaw );
-		if( authnResponse?.SessionToken is not null ) {
-			return new AuthenticateResponse.Success( authnResponse.SessionToken );
+		try {
+			var request = new VerifyMfaChallengeResponseRequest(
+				StateToken: stateToken,
+				PassCode: challengeResponse );
+
+			var resp = await _httpClient.PostAsJsonAsync(
+				$"authn/factors/{factorId}/verify",
+				request,
+				SourceGenerationContext.Default.VerifyMfaChallengeResponseRequest );
+
+			var authnResponse = await JsonSerializer.DeserializeAsync(
+				await resp.Content.ReadAsStreamAsync(),
+				SourceGenerationContext.Default.AuthenticateResponseRaw );
+			if( authnResponse?.SessionToken is not null ) {
+				return new AuthenticateResponse.Success( authnResponse.SessionToken );
+			}
+			throw new BmxException( "Okta MFA challenge response session token null" );
+		} catch( Exception ex ) {
+			throw new BmxException( "Error verifying Okta MFA challenge response", ex );
 		}
-		throw new BmxException( "Error verifying Okta MFA challenge response" );
 	}
 
 	async Task<OktaSession> IOktaApi.CreateSessionAsync( string sessionToken ) {
-		var resp = await _httpClient.PostAsJsonAsync(
-			"sessions",
-			new CreateSessionRequest( sessionToken ),
-			SourceGenerationContext.Default.CreateSessionRequest );
-
-		var session = await JsonSerializer.DeserializeAsync(
-			await resp.Content.ReadAsStreamAsync(),
-			SourceGenerationContext.Default.OktaSession );
-
-		return session ?? throw new BmxException( "Error creating Okta session" );
+		try {
+			var resp = await _httpClient.PostAsJsonAsync(
+				"sessions",
+				new CreateSessionRequest( sessionToken ),
+				SourceGenerationContext.Default.CreateSessionRequest );
+
+			var session = await JsonSerializer.DeserializeAsync(
+				await resp.Content.ReadAsStreamAsync(),
+				SourceGenerationContext.Default.OktaSession );
+			return session ?? throw new BmxException( "Creating Okta Session returned null" );
+		} catch( Exception ex ) {
+			throw new BmxException( "Error creating Okta session", ex );
+		}
 	}
 
 	async Task<OktaApp[]> IOktaApi.GetAwsAccountAppsAsync() {
-		var apps = await _httpClient.GetFromJsonAsync(
-			"users/me/appLinks",
-			SourceGenerationContext.Default.OktaAppArray );
-
-		return apps?.Where( app => app.AppName == "amazon_aws" ).ToArray()
-			?? throw new BmxException( "Error retrieving AWS accounts from Okta" );
+		try {
+			var apps = await _httpClient.GetFromJsonAsync(
+				"users/me/appLinks",
+				SourceGenerationContext.Default.OktaAppArray );
+
+			return apps?.Where( app => app.AppName == "amazon_aws" ).ToArray()
+				?? throw new BmxException( "Retrieving AWS accounts from Okta returned null" );
+		} catch( Exception ex ) {
+			throw new BmxException( "Error retrieving AWS accounts from Okta", ex );
```

Comment:
I think these extra catches could be misleading. If I change the route [here](https://github.com/Brightspace/bmx/blob/3e365d3e5d48abc8b3f9ec80b3bba868d460d3eb/src/D2L.Bmx/Okta/OktaApi.cs/#L129) from `sessions` to `sessionss` I'd expect to see the exception message mentioning a problem to create an okta session. Instead though I get this message here `Error retrieving AWS accounts from Okta`

Reply:
Maybe we need to also pass up the caught exception message if it's a bmx exception?